adafruit_max31855.py

```diff
@@ -0,0 +1,72 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2017 Radomir Dopieralski for Adafruit Industries.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+"""
+``adafruit_max31855``
+=====================
+
+This is a CircuitPython driver for the Maxim Integrated MAX31855 thermocouple
+amplifier module.
+
+"""
+
+import ustruct
+
+from adafruit_bus_device.spi_device import SPIDevice
+
+
+class MAX31855:
+    """
+    Driver for the MAX31855 thermocouple amplifier.
+    """
+
+    def __init__(self, spi, cs):
+        self.spi_device = SPIDevice(spi, cs)
+        self.data = bytearray(4)
+
+    def _read(self, internal=False):
+        with self.spi_device as spi:
+            spi.readinto(self.data)
+        if self.data[3] & 0x01:
+            raise RuntimeError("thermocouple not connected")
+        if self.data[3] & 0x02:
+            raise RuntimeError("short circuit to ground")
+        if self.data[3] & 0x04:
+            raise RuntimeError("short circuit to power")
+        if self.data[1] & 0x01:
+            raise RuntimeError("faulty reading")
+        temp, refer = ustruct.unpack('>hh', self.data)
+        refer >>= 4
+        temp >>= 2
+        if internal:
+            return refer
+        return temp
+
+    @property
+    def temperature(self):
+        """Return measured temperature in degrees Celsius."""
```

Comment:
One small nitpick. This can just be "Thermocouple temperature in degree Celsius." because its a property rather than a function to the outside world.